otter/convergence/steps.py

```diff
@@ -539,3 +552,91 @@ def as_effect(self):
         """
         result = StepResult.LIMITED_RETRY if self.limited else StepResult.RETRY
         return Effect(Constant((result, list(self.reasons))))
+
+
+# ----- Cloud Orchestration Steps -----
+
+
+@implementer(IStep)
+@attr.s
+class CreateStack(object):
+    """
+    A stack must be created.
+
+    :ivar pmap stack_config: Heat launch configuration.
+
+    """
+    stack_config = attr.ib(validator=instance_of(PMap))
+
+    def as_effect(self, append_stack_uuid=append_stack_uuid):
```

Comment:
given how simple of a transform this `append_stack_uuid` operation is, I personally wouldn't bother to parameterize it and override it in the tests -- it just saves you a tiny bit of data declaration in the test to create a config that has the necessary shape to be modified by this function.